framework/src/meshmodifiers/SidesetAroundSubdomainUpdater.C

```diff
@@ -150,6 +163,8 @@ SidesetAroundSubdomainUpdater::finalize()
     for (auto & [elem_id, side] : sent_data)
     {
       _boundary_info.add_side(mesh.elem_ptr(elem_id), side, _boundary_id);
+      _boundary_info.clear_boundary_node_ids();
+      _boundary_info.build_node_list_from_side_list();
```

Comment:
Is this safe? What if the user creates a custom node set (e.g. a single node to suppress rigid body translation). That would just be deleted and not replaced, because not every node set can be represented by an equivalent side set.

Reply:
`MooseMesh` defines `construct_node_list_from_side_list = true` by default.  Is there already another mechanism within Moose that adds to the list for just node sets?

Reply:
But that doesn't _delete_ existing node sets!

Reply:
As long as this doesn't _remove_ any sidesets/nodesets, the clear shouldn't be necessary

Reply:
It will shrink sidesets.

Reply:
What is _it_? The build call or this modifier? Because the build call only adds

Reply:
It seems like creating the nodesets with only a single node is the one-off case since most nodesets are constructed from sidesets.  Wouldn't this call for a seperate fix for this?

Reply:
Most nodesets in MOOSE are constructed from sidesets, but not all, and removing any other nodesets is definitely a bug.

Most nodesets in other software are specifically discrete sets of boundary points unrelated to sets of boundary sides.  MOOSE is only an exception here because back in 2009 I still reviewed libMesh contributions with a mindset of "does this new API do what it says it does" rather than also wondering "what XY Problem do I need to address before it's too late".

You might consider removing node BCIDs on a side's nodes whenever you remove that BCID on that side; then you could rerun `build_node_list_from_side_list()` to restore any overzealous removals without having to clear anything first.

Reply:
The issue that remains is that if users explicitly set `construct_node_list_from_side_list = false` in their mesh block, they will get an unexpected construction of those node sets later on by this UO. But Roy's suggestion is a good compromise between being correct and easy to implement....